aws_handler/lib/aws_handler.ex

```diff
@@ -0,0 +1,204 @@
+defmodule AwsHandler do
+  @moduledoc """
+  Documentation for `AwsHandler`.
+  """
+
+  @doc """
+  Uploads a file to the AWS S3.
+
+  ## Parameters
+
+  - file_path: The path where the file that you want to upload is.
+  - bucket_name: The name of the bucket where you want to upload the file.
+  - storage_path: The path where you want to store the file.
+
+  ## Examples
+
+      iex> AwsHandler.upload_file("README.md", "elixir-trainings", "Oriana/Read.md")
+      {:ok, "File uploaded successfully"}
+
+  """
+  def upload_file(file_path, bucket_name, storage_path) do
+    case File.read(file_path) do
+      {:ok, binary} -> upload_file_s3(binary, bucket_name, storage_path)
+      {:error, _reason} -> {:error, "File not found."}
+    end
+  end
+
+  defp upload_file_s3(file, bucket, path) do
+    if exists_bucket?(bucket) do
+      ExAws.S3.put_object(bucket, path, file)
+      |> ExAws.request()
+      |> case do
+        {:ok, _data} -> {:ok, "File uploaded successfully"}
+        {:error, {http_error, code, _}} -> {:error, "#{http_error} #{code}"}
```

Comment:
Ademas de retornar el error que vas a usar arriba, lo cual es super correcto, te recomiendo logguearlo, y a su vez, retornarlo, ejemplo

{:error, reason} ->
Logger.error("[ModuleName] Upload not successfull for #{bucket}, reason: #{inspect(reason)}")
{:error, "#{http_error} #{code}"}